src/pulsar_producer.erl

```diff
@@ -517,19 +517,6 @@ is_replayq_durable(#{replayq_offload_mode := true}, _Q) ->
 is_replayq_durable(_, Q) ->
     not replayq:is_mem_only(Q).
 
-parse({incomplete, Bin}, State) ->
-    {keep_state, State#{last_bin := Bin}};
-parse({Cmd, <<>>}, State) ->
-    handle_response(Cmd, State#{last_bin := <<>>});
```

Comment:
The problem was:
if it returns `keep_state_and_data` from here, `gen_statem` will restore the old state.